### PR TITLE
SECURITY.md: add section with coding practices

### PR DESCRIPTION
**Pull request by Rugvip**
Want to get these practices/pitfalls documented somewhere, figured `SECURITY.md` will do for now.

### CODE REVIEW

**Review by jhaals — APPROVED** [review]
Great to see this! 🏅 

I think it would be even better with both good and bad practises in both examples

**Comment by Rugvip** [comment]
Yeah true, will add that in 👍

**Comment by freben** [comment]
Pssh you knew you wanted to add that yourself, but wanted to snag the 10k PR!

**Comment by Rugvip** [comment]
@freben nah this one was waiting for a while, wanted to give it some extra attention x)

File: SECURITY.md [hunk 1]
```diff
@@ -36,3 +36,79 @@ There are many situations where a vulnerability does not affect a particular dep
...
+
+function writeTemporaryFile(tmpDir: string, name: string, content: string) {
+  const filePath = join(tmpDir, name);
+  await fs.writeFile(filePath, content);
+}
+```
+
+If the `name` of the file is controlled by the user, they can for example enter `../../../../etc/hosts` as the name of the file. This can lead to a file being written outside the intended directory, which in turn can be used to inject malicious code or other form of attacks.
+
+The recommended solution to this is to use `resolveSafeChildPath` from `@backstage/backend-common` to resolve the file path instead. It makes sure that the resolved path does not fall outside the provided directory. If you simply what to validate whether a file path is safe, you can use `isChildPath` instead.
```
Comment (freben) [inline SECURITY.md:64]:
```suggestion
The recommended solution to this is to use `resolveSafeChildPath` from `@backstage/backend-common` to resolve the file path instead. It makes sure that the resolved path does not fall outside the provided directory. If you simply want to validate whether a file path is safe, you can use `isChildPath` instead.
```

File: SECURITY.md [hunk 2]
```diff
@@ -36,3 +36,79 @@ There are many situations where a vulnerability does not affect a particular dep
...
+
+No matter how trivial it may seem, always use `.json(...)`. It reduces the risk that a future refactoring introduces vulnerabilities:
+
+```ts
+res.send({ ok: true }); // BAD
+
+res.json({ ok: true }); // GOOD
+```
+
+If you absolute must return a string with `.send(...)`, use an explicit and secure `Content-Type`:
```
Comment (freben) [inline SECURITY.md:106]:
```suggestion
If you absolutely must return a string with `.send(...)`, use an explicit and secure `Content-Type`:
```

**Review by freben — APPROVED** [review]

**Comment by mufaddal7** [comment]
#10000 🎉

**Pull request merged**